source/MaterialXTest/MaterialXCore/Node.cpp

```diff
@@ -651,10 +663,32 @@ TEST_CASE("Node Definition Creation", "[nodedef]")
             }
         }
 
-        // Modify the existing nodegraph
+        // Add new version 
+        const std::string VERSION2 = "2.0";
         newGraphName = mx::EMPTY_STRING;
-        nodeDef = doc->addNodeDefFromGraph(graph, newNodeDefname + "0", graph->getName() + "0", newGraphName, "adjustment");
+        nodeDef = doc->addNodeDefFromGraph(graph, newNodeDefName + "2", NODENAME, VERSION2, GROUP, newGraphName);
+        nodeDef->setDefaultVersion(true);
         REQUIRE(nodeDef != nullptr);
+
+        std::vector<mx::NodeDefPtr> matchingNodeDefs;
+        for (auto docNodeDef : doc->getNodeDefs())
+        {
+            if (docNodeDef->getNodeString() == NODENAME)
+            {
+                matchingNodeDefs.push_back(docNodeDef);
+            }
+        }
+        bool findDefault = false;
+        for (auto matchingDef : matchingNodeDefs)
+        {
+            if (matchingDef->getDefaultVersion())
+            {
+                findDefault = true;
```

Comment:
Should you add a check that findDefault is false when entering this if statement? It should only be hit once is my understanding.

Reply:
It currently breaks after the first found. But true, I have no check if it's specified > 1 time. I'll add a test.

Reply:
No sure I understand.. how could it not be false when entering the if statement? Since you break the loop if you enter the if statement.

Reply:
True.